docs/content/docs/connectors/datastream/dynamodb.md

```diff
@@ -23,16 +23,143 @@ KIND, either express or implied.  See the License for the
+
+// Specify watermarking strategy and the name of the DynamoDB Streams Source operator.
+// Specify return type using TypeInformation.
+// Specify also UID of operator in line with Flink best practice.
+val cdcEventsWithEventTimeWatermarks = env.fromSource(dynamoDbStreamsSource, WatermarkStrategy.<String>forMonotonousTimestamps().withIdleness(Duration.ofSeconds(1)), "DynamoDB Streams source")
+  .uid("custom-uid")
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+The above is a simple example of using the `DynamoDbStreamsSource`.
+- The DynamoDB Stream being read from is specified using the stream ARN.
+- Configuration for the `Source` is supplied using an instance of Flink's `Configuration` class.
+  The configuration keys can be taken from `AWSConfigOptions` (AWS-specific configuration) and `DynamodbStreamsSourceConfigConstants` (DynamoDB Streams Source configuration).
+- The example specifies the starting position as `TRIM_HORIZON` (see [Configuring Starting Position](#configuring-starting-position) for more information).
+- The deserialization format is as `SimpleStringSchema` (see [Deserialization Schema](#deserialization-schema) for more information).
+- The distribution of shards across subtasks is controlled using the `UniformShardAssigner`  (see [Shard Assignment Strategy](#shard-assignment-strategy) for more information).
+- The example also specifies an increasing `WatermarkStrategy`, which means each record will be tagged with event time specified using `approximateCreationDateTime`.
+  Monotonically increasing watermarks will be generated, and subtasks will be considered idle if no record is emitted after 1 second.
+
+### Configuring Starting Position
+
+To specify the starting position of the `DynamodbStreamsSource`, users can set the `DynamodbStreamsSourceConfigConstants.STREAM_INITIAL_POSITION` in configuration.
+- `LATEST`: read all shards of the stream starting from the latest record.
+- `TRIM_HORIZON`: read all shards of the stream starting from the earliest record possible (data is trimmed by DynamoDb after 24 hours).
+
+### Deserialization Schema
+
+The `DynamoDbStreamsSource` provides the `DynamoDbStreamsDeserializationSchema<T>` interface to allow users to implement their own
+deserialization schema to convert DynamoDB change data capture events into custom event types.
+
+The `DynamoDbStreamsDeserializationSchema<T>#deserialize` method takes in an instance of `Record` from the DynamoDB model.
+The `Record` can contain different content, depending on the configuration of the DynamoDB Stream. See [AWS docs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Streams.html#Streams.Enabling) for more information.
+
+### Shard Assignment Strategy
+
+The `UniformShardAssigner` allocates shards to parallel subtasks of the `Source` operator based on which parallel subtask has the lowest number of allocated shards.
+The idea is to distribute change data capture events as evenly as possible across parallel subtasks.
```

Comment:
This is not superclear. I would start with explaining what is the main effect of this assigner, then explain how it works and why. Something like this

> The `UniformShardAssigner` allocated the shards of the DynamoDB Stream evenly across the parallel subtasks of the source operator. DynamoDB Stream are ephemeral and are created and deleted automatically, as needed. `UniformShardAssigner` allocates new shards to the subtask with the lowest number of allocated shards.

Reply:
Why is "`Source`" shown as code? In this case you can say "source operator"

Reply:
Question: does the connector follow the lineage of the shards? 
There are implications with the ordering, in particular when you start from `TRIM_HORIZON`.
If the source does not follow the shard lineage, change event may be replayed out of order. This is something that should be highlighted

Reply:
I am also putting here because it's related to ordering that is an important subject and we need to be explicit on this in docs.

I assume DDBS shards by record primary key, otherwise it could not guarantee any order, but I can't find it in DDB documentation. 
If that is the case, we should mention the fact because Flink will get change events in order per PK (as long as the source is consuming the latest changes or also when replaying from TRIM_HORIZON, if the source follows the lineage of the shards)

Reply:
> Question: does the connector follow the lineage of the shards?
There are implications with the ordering, in particular when you start from TRIM_HORIZON.
If the source does not follow the shard lineage, change event may be replayed out of order. This is something that should be highlighted


Yes, great callouts. The connector follows shard ordering. Have added a subsection on `### Event Ordering`.